### ISSUE TITLE
[Feature] Search Input handling

### ISSUE THREAD

**Issue by dikehprosper** [creativecommons/search#93]
## Problem
 At the moment there isn't a clear path for handling a search input .

## Description
this targets the functionality of search input not the design as the design can be made to suit CC global components.
suggesting the search input handling......
1.  the user type in the search word,
2.   the user chooses a search option (should be highlighted) and
3.  clicks on the search button.
4.  which then gives access through the search services chosen.
  the visual example below...

![photo_2022-10-14_13-45-09](https://user-images.githubusercontent.com/103826657/195909061-b21998f0-8295-4af5-9064-b50b31ba6a75.jpg)

## Implementation

- [x] I would be interested in implementing this feature.

**Comment by possumbilities** [comment]
Thank you for pointing this out! I think this is a reasonable UX problem. I'm interested in any solutions anyone has for how to improve this flow to make it a bit more intuitive?

**Comment by dipanshuraz** [comment]
Hi everyone, I'm outreachy applicant, I'm interested to pick this up.

**Comment by ImaJin14** [comment]
@dipanshuraz this work still needs the status ready for work before it can be worked on

**Comment by ImaJin14** [comment]
@dipanshuraz this work still needs the status ready for work before it can be worked on

**Comment by dikehprosper** [comment]
> Thank you for pointing this out! I think this is a reasonable UX problem. I'm interested in any solutions anyone has for how to improve this flow to make it a bit more intuitive?

A solution I would suggest is allowing a user to select his/her search engine first, then type in search-word, then click on the provided search button for answers.

**Comment by dikehprosper** [comment]
> Hi everyone, I'm outreachy applicant, I'm interested to pick this up.



> Hi everyone, I'm an outreachy applicant, I'm interested to pick this up.

hi @dipanshuraz I already indicated interest tho but we can work on it together when approved since you have shown interest too

**Comment by ketchanji** [comment]
hello every am working on the search button at suppon to be on. 

https://oldsearch.creativecommons.org/
https://github.com/creativecommons/search

**Comment by dikehprosper** [comment]
Hello @ketchanji it hasn’t been approved yet to do so……and there are people already proposing to do it…to avoid code conflict I would advice you ask to join

**Comment by ketchanji** [comment]
so what am i to do now ?

**Comment by ketchanji** [comment]
>sir @dikehprosper  I  am already done with the button its left just to push sir.

**Comment by possumbilities** [comment]
Hello everyone, just a reminder that people shouldn't be working on this yet since it doesn't have a `status: ready for work` on it. 

From the [Contribution Guidelines](https://opensource.creativecommons.org/contributing-code/#finding-an-issue):

> Issues not ready for work:
The following tags mark issues that are not open for community contribution:
🚧 status: blocked: Blocked by other work that needs to be done first
🧹 status: ticket work required: Needs additional work before it is ready to be taken up
🚦 status: awaiting triage: Has not been triaged by a maintainer
Do not work on these.

If you come across an Issue that isn't ready for work yet, I'd advise that you move onto generating new Issues, or doing work on Issues that are available to work on. An Issue may not be ready for work for a number of reasons and it may take a period of time before it is ready, if ever.

**Comment by ketchanji** [comment]
please sir @possumbilities in that case its difficult for us cause am loocking for projects i can contribute too but am  not seeing any and also am i don't know were i can fine some ongoing projects to work on.

**Comment by dikehprosper** [comment]
@ketchanji you can create an issue to work on an then wait for it to be approved and work on it

**Comment by ketchanji** [comment]
ah ok sir @dipanshuraz . should i still wait or i should create a different issue,

**Comment by JennySimen** [comment]
@dikehprosper here is a suggestion on how to direct users on using the search functionality.

A starting point could be to add some directions on how to go about searching. 

For example:

Let's say replacing the search input placeholder with `Search using ccMixter engine (selected engine)`. 

Then adding a text below the search input saying `You can choose another search engine below`.

Well! the messages could be better formatted but this way, users know how to go about choosing search engines.

**Comment by ketchanji** [comment]
> > please sir @possumbilities in that case its difficult for us cause am loocking for projects i can contribute too but am not seeing any and also am i don't know were i can fine some ongoing projects to work on.
> 
> It's ma'am actually, not sir :)
> 
> As @dikehprosper offered there are many ways to contribute. Adding Issues is one way, participating in a discussion on an issue with ideas; there are many. Submitting a PR for an Issue is not the only or primary way to contribute :)

okk sir thanks you i understand.

**Comment by ketchanji** [comment]
> @ketchanji you can create an issue to work on and then wait for it to be approved and work on it

@dikehprosper sir can i add this to my contribution record as assisting in idea on how to come about the solution?

**Comment by dikehprosper** [comment]
@ketchanji there is an ongoing discussion on how the search input can be handled….I would advice you relay a tangible suggestion…and then add it to your contribution.

**Comment by dikehprosper** [comment]
> 

This is good @JennySimen

**Comment by ImaJin14** [comment]
Best case I find here will be to remove the keyboard search option and move it completely to engine selection.... Or we make the engine selection standard and the search button the only search option


That means if a user inputs a search query they can search by choosing their search engine and not clicking enter ..... 

Or

They select their desired search engine then input their search query and press enter

**Comment by ketchanji** [comment]
> @ketchanji there is an ongoing discussion on how the search input can be handled….I would advice you relay a tangible suggestion…and then add it to your contribution.

I would suggest we redesign the ui such that
There are 3 actions to be done when performing a search and probably only 2 will be required

Firstly, the user enters their search query in an input field
Next to this input field is a drop down menu from where the user selects the search engine they would prefer to query from
lastly, the user clicks on the search button where their query will be submitted alongside the engine of preference.

The UI will be positioned on a single block manner to make the UX straight forward.

Then in the section previously used for selecting the search engine, we could place cards in a row that  informs the user of each search engine and its appropriate/suggested cases

If no query is specified, no action is done and if no engine is selected, a default engine could be used in its place.

**Comment by PreciousOritsedere** [comment]
I would suggest that the search input should be used to type in the name of a search engine.
For instance, when more search engines are added to the website, it might get a like tacky to navigate or find the specific search engine you want. 
So in such case, you can just type in the name of the search engine(or something similar) and the when you click on search, all search engines related to the word in the search input will be displayed on the screen.

That way, you can easily just find the one you need and click on it.

**Comment by possumbilities** [comment]
Resolved in new site merge: https://github.com/creativecommons/search/pull/206
